kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/WindowDoc.java

```diff
@@ -41,6 +43,17 @@ public class WindowDoc {
   public WindowDoc() {
   }
 
+  @BsonCreator
```

Comment:
do we need to do the same for `KVDoc`? I can't remember if you said that's going to be a separate PR or not. Also for `KVDoc` I noticed there's `@BsonId` annotation on the `id` field, is that necessary/needed?

Reply:
I did some testing on KVDoc independently and it looks like `@BsonId` isn't doing what we want it to. I wanted to see what the feedback was on this Window stuff before applying the same logic to the `KVDoc` (since it's already in use in production).

Reply:
note: none of the Mongo stuff is in production yet, the KV store that we have in prod is based on Scylla. So now is the time to make any changes we want/need